app/src/main/java/org/p2p/wallet/debug/settings/DebugSettingsDeviceInfoMapper.kt

```diff
@@ -0,0 +1,108 @@
+            val displayMetrics: DisplayMetrics = resources.displayMetrics
+            val densityBucket = getDensityString(displayMetrics)
+            val deviceMake = truncateAt(Build.MANUFACTURER, 20)
+            val deviceModel = truncateAt(Build.MODEL, 20)
+            val deviceResolution = displayMetrics.heightPixels.toString() + "x" + displayMetrics.widthPixels
+            val deviceDensity = displayMetrics.densityDpi.toString() + "dpi (" + densityBucket + ")"
+            val deviceRelease = Build.VERSION.RELEASE
+            val deviceApi = Build.VERSION.SDK_INT.toString()
+            createRecord("densityBucket", densityBucket)
+            createRecord("deviceMake", deviceMake)
+            createRecord("deviceModel", deviceModel)
+            createRecord("deviceResolution", deviceResolution)
+            createRecord("deviceDensity", deviceDensity)
+
+            appendBreakLine()
+
+            createRecord("deviceRelease", deviceRelease)
+            createRecord("deviceApi", deviceApi)
+        }
+        return listOf(
+            SettingsRow.Info(
+                R.string.debug_settings_device_info,
+                subtitle = deviceValues
+            )
+        )
+    }
+
+    fun mapCiInfo(): List<SettingsRow> {
+        val ciValues = buildString {
+            createApiKeyRecord("amplitudeKey", BuildConfig.amplitudeKey)
+            createApiKeyRecord("intercomApiKey", org.p2p.core.BuildConfig.intercomApiKey)
+            createApiKeyRecord("intercomAppId", org.p2p.core.BuildConfig.intercomAppId)
+            createApiKeyRecord("moonpayKey", org.p2p.core.BuildConfig.moonpayKey)
+            createApiKeyRecord("moonpaySanbdoxKey", org.p2p.core.BuildConfig.moonpaySandboxKey)
+            createApiKeyRecord("rpcPoolApiKey", org.p2p.core.BuildConfig.rpcPoolApiKey)
+
+            appendBreakLine()
+
+            createFlagRecord("CRASHLYTICS_ENABLED", org.p2p.core.BuildConfig.CRASHLYTICS_ENABLED)
+        }
+        return listOf(
+            SettingsRow.Info(
+                R.string.debug_settings_ci_info,
+                subtitle = ciValues
+            ),
+        )
+    }
+
+    fun mapAppInfo(): List<SettingsRow> {
+        return listOf(
+            SettingsRow.Title(R.string.debug_settings_app_info, isDivider = true),
+            SettingsRow.Section(
+                titleResId = R.string.settings_app_version,
+                subtitle = "${BuildConfig.BUILD_TYPE}-${BuildConfig.VERSION_NAME}",
+                iconRes = R.drawable.ic_settings_app_version
+            )
+        )
+    }
+
+    private fun StringBuilder.createApiKeyRecord(apiKeyName: String, apiKey: String) {
+        append("$apiKeyName = ")
+        append("***")
+        append(apiKey.removeRange(startIndex = 0, endIndex = apiKey.length - 3))
+        appendBreakLine()
+    }
+
+    private fun StringBuilder.createFlagRecord(flagName: String, flagValue: Boolean) {
+        append("$flagName = $flagValue")
+        appendBreakLine()
+    }
```

Comment:
These guys are not creating anything, just appending

```suggestion
    private fun StringBuilder.appendApiKeyRecord(apiKeyName: String, apiKey: String) {
        append("$apiKeyName = ")
        append("***")
        append(apiKey.removeRange(startIndex = 0, endIndex = apiKey.length - 3))
        appendBreakLine()
    }

    private fun StringBuilder.appendFlagRecord(flagName: String, flagValue: Boolean) {
        append("$flagName = $flagValue")
        appendBreakLine()
    }
```